lib/src/main/java/io/ably/lib/types/MessageAnnotations.java

```diff
@@ -0,0 +1,155 @@
+package io.ably.lib.types;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import io.ably.lib.util.Log;
+import org.msgpack.core.MessageFormat;
+import org.msgpack.core.MessagePacker;
+import org.msgpack.core.MessageUnpacker;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+
+/**
+ * Contains information about annotations associated with a particular message.
+ */
+public class MessageAnnotations {
+
+    private static final String TAG = MessageAnnotations.class.getName();
+
+    private static final String SUMMARY = "summary";
+
+    /**
+     * A summary of all the annotations that have been made to the message. Will always be
+     * populated for a message.annotations.summary, and may be populated for any other type (in
+     * particular a message retrieved from REST history will have its latest summary
+     * included).
+     * The keys of the map are the annotation types. The exact structure of the value of
+     * each key depends on the aggregation part of the annotation type, e.g. for a type of
+     * reaction:distinct.v1, the value will be a DistinctValues object. New aggregation
+     * methods might be added serverside, hence the 'unknown' part of the sum type.
+     */
+    public Summary summary;
+
+    public MessageAnnotations() {
+        // Protocol v4: Summary is non-nullable, initialize with empty summary
+        this.summary = new Summary(new HashMap<>());
+    }
+
+    public MessageAnnotations(Summary summary) {
+        // Protocol v4: Ensure summary is never null
+        this.summary = summary != null ? summary : new Summary(new HashMap<>());
+    }
+
+    void writeMsgpack(MessagePacker packer) throws IOException {
+        int fieldCount = 0;
+        if (summary != null) ++fieldCount;
+
+        packer.packMapHeader(fieldCount);
+
+        if (summary != null) {
+            packer.packString(SUMMARY);
+            summary.write(packer);
+        }
+    }
+
+    MessageAnnotations readMsgpack(MessageUnpacker unpacker) throws IOException {
```

Comment:
```suggestion
    static MessageAnnotations read(MessageUnpacker unpacker) throws IOException {
```
`read` methods used for decoding msgpack/json across all `message classes` are static.
This will automatically make sure we use  `MessageAnnotations` constuctor with non-null summary.